File: movies money/code.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char titulo[100];
    float lucro;
} Filme;

void classificarFilmes(Filme* filmes, int num_filmes) {
    int i, j;
    for (i = 0; i < num_filmes - 1; i++) {
        for (j = 0; j < num_filmes - i - 1; j++) {
            if (filmes[j].lucro < filmes[j+1].lucro) {
                Filme temp = filmes[j];
                filmes[j] = filmes[j+1];
                filmes[j+1] = temp;
            }
        }
    }
}

void exibirClassificacao(Filme* filmes, int num_filmes) {
    printf("%-4s%-40s%-12s\n", "Rank", "Título", "Lucro (em bilhões USD)");
    printf("===============================================\n");
    for (int i = 0; i < num_filmes; i++) {
        printf("%-4d%-40s%-12.2f\n", i + 1, filmes[i].titulo, filmes[i].lucro);
    }
}

int main() {
    // Dados fictícios de filmes
    Filme filmes[] = {
        { "Avatar", 2.847 },
        { "Vingadores: Ultimato", 2.798 },
        { "Titanic", 2.194 },
        { "Star Wars: Episódio VII - O Despertar da Força", 2.068 },
        { "Vingadores: Guerra Infinita", 2.048 }
    };

    int num_filmes = sizeof(filmes) / sizeof(filmes[0]);

    // Classificar filmes pelo lucro
    classificarFilmes(filmes, num_filmes);

    // Exibir tabela de classificação
    exibirClassificacao(filmes, num_filmes);

    return 0;
}

File: movies money/code.py
class Filme:
    def __init__(self, titulo, lucro):
        self.titulo = titulo
        self.lucro = lucro

def classificar_filmes(filmes):
    filmes.sort(key=lambda x: x.lucro, reverse=True)

def exibir_classificacao(filmes):
    print("{:<4}{:<40}{:<20}".format("Rank", "Título", "Lucro (em bilhões USD)"))
    print("======================================================")
    for i, filme in enumerate(filmes, start=1):
        print("{:<4}{:<40}{:<20.2f}".format(i, filme.titulo, filme.lucro))

# Dados fictícios de filmes
filmes = [
    Filme("Avatar", 2.847),
    Filme("Vingadores: Ultimato", 2.798),
    Filme("Titanic", 2.194),
    Filme("Star Wars: Episódio VII - O Despertar da Força", 2.068),
    Filme("Vingadores: Guerra Infinita", 2.048)
]

# Classificar filmes pelo lucro
classificar_filmes(filmes)

# Exibir tabela de classificação
exibir_classificacao(filmes)
